src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller_Commit.cs

```diff
@@ -122,14 +123,32 @@ private void Commit(CompletionItem item, TextChange textChange, Model model, cha
 
             var document = this.SubjectBuffer.CurrentSnapshot.GetOpenDocumentInCurrentContextWithChanges();
             var formattingService = document.GetLanguageService<IEditorFormattingService>();
-            if (formattingService != null &&
-                (item.ShouldFormatOnCommit || (commitChar != null && formattingService.SupportsFormattingOnTypedCharacter(document, commitChar.GetValueOrDefault()))))
+
+            var commitCharTriggersFormatting = commitChar != null && 
+                    (formattingService?.SupportsFormattingOnTypedCharacter(document, commitChar.GetValueOrDefault()) 
+                     ?? false);
+
+            if (formattingService != null && item.ShouldFormatOnCommit || commitCharTriggersFormatting)
```

Comment:
ah, yeah. It was intentional that the && applies to the first 2 sub expressions. I initially had it as a separate local, and then inlined the temp :) I'll make it look r**eally intentiona**l. thanks.